Add patronictl -k/--insecure flag and suport for restapi cert

wilfriedroset opened the pull request:
Fixes https://github.com/zalando/patroni/issues/785

CyberDem0n commented on patroni/ctl.py:
```diff
@@ -128,17 +130,28 @@ def auth_header(config):
     url = '{0}://{1}/{2}'.format(url_parts.scheme, url_parts.netloc, endpoint)
-
+    # Retrieve certfile from configfile in restapi section if any
+    cert = (click.get_current_context().obj.get('restapi', {}).get('certfile'),
```
Doc http://docs.python-requests.org/en/master/user/advanced/#client-side-certificates tells that `cert` is used for client-side certificates, but here you are presumably trying to use server-side.

Alright, let's assume that this is a special patronictl config. Even in this case there is a problem: `requests` module expect to get either a `string` (filename containing cert+key) or a tuple. Not sure how it will work if it will get for example `('cert', None)`, because apparently [keyfile](https://patroni.readthedocs.io/en/latest/SETTINGS.html?highlight=keyfile#rest-api) is optional: https://docs.python.org/2/library/ssl.html#ssl.SSLContext.load_cert_chain

wilfriedroset replied:
Indeed, I will remove this part.

wilfriedroset commented:
I was first going for the configuration file part because I think that we will have to use `-k` a lot. But by doing so we would need to duplicate the restapi certfile/keyfile part to the ctl part or use some kind of interpolation.

CyberDem0n commented:
> I was first going for the configuration file part because I think that we will have to use -k a lot.

If you are using self-signed certificates, especially if every node has different certificates, you'll definitely have to use `-k` a lot.
Taking into account that you can run `patronictl` on any host, including your computer, I think it make sense to have a special section for patronictl in the config.

Think about following use-case: certificates on the patroni nodes are signed by the same certificate authority. Patronictl can use `cacert` to verify node certificates. Sure, such cacert can go into `restapi`.
Another use-case would be: we want to use client certificates to authenticate patronictl. It definitely requires a special section (`ctl`) in the config.

> we would need to duplicate the restapi certfile/keyfile part to the ctl part or use some kind of interpolation.

It is very easy to avoid duplication. If `ctl.cacert` is not there, try to use `restapi.cacert` or `restapi.certfile`.

If the `-k` is specified, it should just change the "virtual" config: `ctx.obj['ctl']['insecure'] = True` or `ctx.obj['ctl']['verify'] = False`
You can do it just in one place, in the `def ctl(ctx, config_file, dcs):` function. And later use the result in the `request_patroni`. It will simplify the code, because you don't need to pass around all `insecure` arguments.

wilfriedroset commented:
What do you think about this version?
`ctx.obj` is updated during `def ctl`.
I've added a graceful support for configuration file which does not contain the `ctl` section.
The context is retrieve using `click.get_current_context()` and we avoid double definition for certfile/cacert using your recommendation.

About travis build I don't understand the failure.
This build pass: https://travis-ci.org/zalando/patroni/builds/421115034?utm_source=github_status&utm_medium=notification
This one doesn't: https://travis-ci.org/zalando/patroni/builds/421125607?utm_source=github_status&utm_medium=notification

The difference between the two is that I've updated the documentation.

CyberDem0n commented:
I'll look on it a little bit later.

Acceptance tests sometimes are not stable due to the slowness of travis :(
I've re-triggered it, usually it helps.

CyberDem0n commented:
Merged, thank you!
